### PR TITLE
 Add Protobuf schemas for existing PySyft messages and types

### PR DESCRIPTION
**Pull request by karlhigley**
A first stab at following through on the [Protobuf RFC](https://github.com/OpenMined/rfcs/pull/6).

This is a fairly direct translation of the existing serialization
into Protobuf, retaining as much of the current flexibility as possible
while capturing explicit message schemas.

### CODE REVIEW

**Comment by karlhigley** [comment]
BTW, I'm thinking of using [Buf](https://buf.build/) for managing versions, linting, and breaking change detection.

File: pysyft/v1/object_message.proto [hunk 1]
```diff
@@ -0,0 +1,9 @@
+syntax = "proto3";
+
+package pysyft.v1;
+
+message ObjectMessage {
+    // The key thing to work out here is how to serialize tensors in Protobuf.
+    // Tensorflow must already be doing this, so might be able to draw inspiration
+    // from existing code.
```
Comment (jvmancuso) [inline pysyft/v1/object_message.proto]: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/tensor.proto

**Comment by jvmancuso** [comment]
I think you'll also want to specify what kind of base tensor is being sent (e.g. Torch or TensorFlow)

**Comment by jvmancuso** [comment]
in general, it might be good to check out the `AbstractTensor.send` method, all of the different functions it calls, and finally the various serde simplifiers/detailers for anything that gets packed up as an ObjectMessage. I'm thinking that would be a comprehensive list of most of the info needed for  `x.send()`

**Comment by jvmancuso** [comment]
lastly, keep in mind it might make sense to subtype the ObjectMessage

**Comment by karlhigley** [comment]
@jvmancuso Thanks for the tips! I'll start looking into tensor serialization once I finish picking some of the lower hanging fruit (e.g. PointerTensors.)

**Comment by karlhigley** [comment]
Okay, at this point, I've (mostly) enumerated the Operation and PlanCommand messages currently in use, but haven't yet created separate schemas for them. The `buf` linter doesn't like this directory structure, since it (1) doesn't consider v0 a valid version, and (2) thinks all packages should end with a version number (i.e. a versioned directory).

**Comment by karlhigley** [comment]
I'm somewhere around a third of the way through the [various uses of the `Operation` message](https://github.com/karlhigley/PySyft-Proto/pull/1/files#diff-5be0a33160360c4dc13fb73ce97d1a5fR14-R32) I could find. The ones that are left are rather difficult to trace through the PySyft code in order to find the various `args` and `kwargs` that get passed around into those messages.

**Comment by karlhigley** [comment]
I think I've gone about as far as I can by reading/searching the code. My next step is probably to instrument the `Operation` class with logging and capture all the arguments that get passed in by the test suite.

**Comment by karlhigley** [comment]
Confirmed that I can capture the messages sent with `Operation` by the tests via some additional logging and PyTest configuration. Will dig into this tomorrow.

**Comment by karlhigley** [comment]
Fleshed out the Syft/Torch types and condensed it together into a single diff. Will take a look at TF types next.

**Comment by karlhigley** [comment]
Superseded by OpenMined/syft-proto#12

**Pull request closed without merging**